metadata/src/audio/item.rs

```diff
@@ -16,98 +17,255 @@ use librespot_core::{
 
 pub type AudioItemResult = Result<AudioItem, Error>;
 
-// A wrapper with fields the player needs
 #[derive(Debug, Clone)]
-pub struct AudioItem {
-    pub id: SpotifyId,
-    pub spotify_uri: String,
-    pub files: AudioFiles,
-    pub name: String,
-    pub duration: i32,
-    pub availability: AudioItemAvailability,
-    pub alternatives: Option<Tracks>,
-    pub is_explicit: bool,
+pub enum AudioItem {
+    Track(Box<Track>),
+    Episode(Box<Episode>),
 }
 
 impl AudioItem {
     pub async fn get_file(session: &Session, id: SpotifyId) -> AudioItemResult {
-        match id.item_type {
-            SpotifyItemType::Track => Track::get_audio_item(session, id).await,
-            SpotifyItemType::Episode => Episode::get_audio_item(session, id).await,
-            _ => Err(Error::unavailable(MetadataError::NonPlayable)),
+        Ok(match id.item_type {
+            SpotifyItemType::Track => AudioItem::Track(Box::new(Track::get(session, &id).await?)),
```

Comment:
Why does clippy complain when you don't box?

Small nitpick, you could probably also return `Self::Track` here.

Reply:
No idea it complained that the Track was much larger then the episode (like twice the size) and said they should be boxed?

Reply:
For reference:

```
warning: large size difference between variants
  --> metadata/src/audio/item.rs:22:5
   |
22 |     Track(Track),
   |     ^^^^^^^^^^^^ this variant is 944 bytes
   |
   = note: `#[warn(clippy::large_enum_variant)]` on by default
note: and the second-largest variant is 464 bytes:
  --> metadata/src/audio/item.rs:23:5
   |
23 |     Episode(Episode),
   |     ^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant
help: consider boxing the large fields to reduce the total size of the enum
   |
22 |     Track(Box<Track>),
   |           ~~~~~~~~~~

warning: `librespot-metadata` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s

```

Reply:
Yep, exactly.

Reply:
Looking at the documentation of that lint, I think that our usual case is for the larger variant, so I would be fine with adding  `#[allow(clippy::large_enum_variant)]` here and remove the boxing.

Reply:
OK I can do that.